packages/emitter-framework/src/java/components/enum.tsx

```diff
@@ -0,0 +1,57 @@
+import { Enum } from "@typespec/compiler";
+import * as jv from  "@alloy-js/java"
+import { mapJoin } from "@alloy-js/core";
+import { TypeExpression } from "../../typescript/index.js";
+
+
+export interface EnumProps {
+  type: Enum;
+}
+
+export function EnumDeclaration({ type }: EnumProps) {
+
+  const members = Array.from(type.members.values());
+
+  let valueType: string | undefined;
+  let isConsistent = true;
+
+  members.forEach((member, index) => {
+    if (member.value !== undefined) {
+      const currentType = typeof member.value === "string" ? "String" : "int";
+
+      if (index === 0) {
+        valueType = currentType;
+      } else if (valueType !== currentType) {
+        isConsistent = false;
+      }
+    }
+  });
+
+  if (!isConsistent) {
+    throw new Error("All enum values must have the same type");
+  }
+
+  let constructor = null;
+  let variable = null;
+  if (valueType) {
+    variable = <jv.Variable type={valueType} name="customValue" private={true}/>;
+    constructor = <jv.Constructor name={type.name} parameters={{"customValue": valueType}}
+                                  children={"this.customValue = customValue;"} private={true}/>;
```

Comment:
Few other feedback for this section:
- `private={true}` can just be `private`
- Passing `children` prop should be done through JSX not as a prop,
do something like this instead.
```
<jv.Constructor>
  this.value = value;
</jv.Constructor>
```
- Where you have `"customValue": valueType`, it can just be `customValue: valueType`